addons/sale/static/src/js/product_configurator_widget.js

```diff
@@ -0,0 +1,212 @@
+ * !!! It should only be used on a product_product or product_template field !!!
+ */
+var ProductConfiguratorWidget = relationalFields.FieldMany2One.extend({
+    resetOnAnyFieldChange: true,
+    events: _.extend({}, relationalFields.FieldMany2One.prototype.events, {
+        'click .o_edit_product_configuration': '_onEditProductConfiguration'
+    }),
+
+    /**
+     * This method will check if the current product_template set on the SO line is configurable
+     * -> If so, we add a 'Edit Configuration' button next to the dropdown.
+     *
+     * @override
+     */
+     start: function () {
+        prom = this._super.apply(this, arguments);
+
+        if (this._isConfigurableProduct()) {
+            this._addConfigurationEditButton();
+        }
+
+        return prom;
+     },
+
+    /**
+     * If current product is configurable,
+     * Show edit button (in Edit Mode) after the product/product_template
+     */
+    _addConfigurationEditButton: function () {
+        var $inputDropdown = this.$('.o_input_dropdown');
+
+        if ($inputDropdown.length !== 0 &&
+            this.$('.o_edit_product_configuration').length === 0) {
+            var $editConfigurationButton = $('<button>', {
+                type: 'button',
+                class: 'fa fa-pencil btn btn-secondary o_edit_product_configuration',
+                tabindex: '-1',
+                draggable: false,
+                'aria-label': _t('Edit Configuration'),
+                title: _t('Edit Configuration')
+            });
+
+            $inputDropdown.after($editConfigurationButton);
+        }
+    },
+
+    /**
+     * Hook to override with _onEditProductConfiguration
+     * to know if edit pencil button has to be put next to the field
+     *
+     * @private
+     */
+    _isConfigurableProduct: function () {
+        return false;
+    },
+
+    /**
+     * Override catching changes on product_id or product_template_id.
+     * Calls _onTemplateChange in case of product_template change.
+     * Calls _onProductChange in case of product change.
+     * Shouldn't be overridden by product configurators
+     * or only to setup some data for further computation
+     * before calling super.
+     *
+     * @override
+     * @param {OdooEvent} ev
+     *   {boolean} ev.data.preventProductIdCheck prevent the product configurator widget
+     *     from looping forever when it needs to change the 'product_template_id'
+     *
+     * @private
+     */
+    _onFieldChanged: function (ev) {
+        var self = this;
+
+        this._super.apply(this, arguments);
+
+        // Hide edit pencil button if product doesn't need to be reconfigured through a wizard.
+        var $inputDropdown = this.$('.o_input_dropdown');
+        if (!self._isConfigurableProduct() &&
+            $inputDropdown.length !== 0 &&
+            this.$('.o_edit_product_configuration').length !== 0){
+            this.$('.o_edit_product_configuration').remove();
```

Comment:
removing is not hiding. You can maybe get rid of this if you render the button in `render`